internal/collector/service_monitor.go

```diff
@@ -0,0 +1,75 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+	log "github.com/ViaQ/logerr/v2/log/static"
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/openshift/cluster-logging-operator/internal/reconcile"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	prometheusCAFile = "/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt"
+)
+
+func NewServiceMonitor(namespace, name string, owner metav1.OwnerReference) *monitoringv1.ServiceMonitor {
+
+	desired := runtime.NewServiceMonitor(namespace, name)
+
+	endpoint := monitoringv1.Endpoint{
+		Port:   MetricsPortName,
+		Path:   "/metrics",
+		Scheme: "https",
+		TLSConfig: &monitoringv1.TLSConfig{
+			CAFile:     prometheusCAFile,
+			ServerName: fmt.Sprintf("%s.%s.svc", name, namespace),
+		},
+	}
+	logMetricExporterEndpoint := monitoringv1.Endpoint{
+		Port:   ExporterPortName,
+		Path:   "/metrics",
+		Scheme: "https",
+		TLSConfig: &monitoringv1.TLSConfig{
+			CAFile:     prometheusCAFile,
+			ServerName: fmt.Sprintf("%s.%s.svc", name, namespace),
+		},
+	}
+
+	labelSelector := metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"logging-infra": "support",
+		},
+	}
+
+	desired.Spec = monitoringv1.ServiceMonitorSpec{
+		JobLabel:  "monitor-collector",
+		Endpoints: []monitoringv1.Endpoint{endpoint, logMetricExporterEndpoint},
+		Selector:  labelSelector,
+		NamespaceSelector: monitoringv1.NamespaceSelector{
+			MatchNames: []string{namespace},
+		},
+	}
+
+	utils.AddOwnerRefToObject(desired, owner)
+
+	return desired
+}
+
+func ReconcileServiceMonitor(k8sClient client.Client, namespace, name string, owner metav1.OwnerReference) {
+	desired := NewServiceMonitor(namespace, name, owner)
+	if err := reconcile.ServiceMonitor(k8sClient, desired); err != nil {
+		log.Error(err, "collector.ReconcileServiceMonitor")
+	}
+}
+
+func RemoveServiceMonitor(k8sClient client.Client, namespace, name string) {
+	serviceMonitor := NewServiceMonitor(namespace, name, metav1.OwnerReference{})
```

Comment:
How about replacing this with an empty resource which just identifies itself with name and namespace?

```suggestion
	serviceMonitor := &monitoringv1.ServiceMonitor{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name,
			Namespace: namespace,
		},
	}
```